### PR TITLE
Use sc.ini file for windows config, fix dcd-server not stopping on Linux

### PR DESCRIPTION
**Pull request by LaurentTreguier**
After the release of 0.6.0, I realized there was that on Linux the `dcd-server` process is never stopped (and can even be spawned multiple times). I had noticed it while debugging, however it worked fine on Windows and https://github.com/Microsoft/vscode/issues/1298 made me think that VSCode was at fault.
It looks like it was because of how external processes are handled, and as they are handled differently on Windows and Linux there was no problem on WIndows...
Now the server is explicitly stopped in the `deactivate()` method instead of making `Server` extend `Disposable` and handing it over to the extension's subscriptions.
In addition, the configuration now uses the `sc.ini` file for Windows and thus is more consistent, and typings are installed through `typings` instead of `tsd` (which takes up 99% of this PR's git diff).

### CODE REVIEW

**Comment by mattiascibien** [comment]
@LaurentTreguier I will merge it ASAP. this is preety neat. Will be marked as 0.6.1 i guess

File: src/extension.ts [hunk 1]
```diff
@@ -55,8 +55,10 @@ export function activate(context: vsc.ExtensionContext) {
             Dscanner.path = dub.packages.get('dscanner').path;
 
             let diagnosticCollection = vsc.languages.createDiagnosticCollection();
-            let lintDocument = (document) => {
-                let dscanner = new Dscanner(document);
+            let lintDocument = (document: vsc.TextDocument) => {
```
Comment (mattiascibien) [inline src/extension.ts:58]: Perfect.

**Comment by mattiascibien** [comment]
I think this is safe to merge. @LaurentTreguier ?

**Comment by LaurentTreguier** [comment]
@mattiascibien I think it is... At least it should be safer than not merging it

**Comment by mattiascibien** [comment]
Done... I guess it should be a good idea to push a new version out on VsCode

**Pull request merged**